index.js

```diff
@@ -44,47 +42,52 @@ class KeyringController extends EventEmitter {
     this.getNetwork = opts.getNetwork
   }
 
-  // Full Update
-  // returns @object state
-  //
-  // Emits the `update` event and
-  // returns a Promise that resolves to the current state.
-  //
-  // Frequently used to end asynchronous chains in this class,
-  // indicating consumers can often either listen for updates,
-  // or accept a state-resolving promise to consume their results.
-  //
-  // Not all methods end with this, that might be a nice refactor.
+  /**
+   * Full Update
+   *
+   * Emits the `update` event and @returns a Promise that resolves to
+   * the current state.
+   *
+   * Frequently used to end asynchronous chains in this class,
+   * indicating consumers can often either listen for updates,
+   * or accept a state-resolving promise to consume their results.
+   *
+   * @returns {Object} The controller state.
+   */
   fullUpdate () {
     this.emit('update', this.memStore.getState())
     return this.memStore.getState()
   }
 
-  // Create New Vault And Keychain
-  // @string password - The password to encrypt the vault with
-  //
-  // returns Promise( @object state )
-  //
-  // Destroys any old encrypted storage,
-  // creates a new encrypted store with the given password,
-  // randomly creates a new HD wallet with 1 account,
-  // faucets that account on the testnet.
+  /**
+   * Create New Vault And Keychain
+   *
+   * Destroys any old encrypted storage,
+   * creates a new encrypted store with the given password,
+   * randomly creates a new HD wallet with 1 account,
+   * faucets that account on the testnet.
+   *
+   * @param {string} password - The password to encrypt the vault with.
+   * @returns {Promise<Object>} A Promise that resolves to the state.
+   */
   createNewVaultAndKeychain (password) {
     return this.persistAllKeyrings(password)
       .then(this.createFirstKeyTree.bind(this))
       .then(this.persistAllKeyrings.bind(this, password))
       .then(this.fullUpdate.bind(this))
   }
 
-  // CreateNewVaultAndRestore
-  // @string password - The password to encrypt the vault with
-  // @string seed - The BIP44-compliant seed phrase.
-  //
-  // returns Promise( @object state )
-  //
-  // Destroys any old encrypted storage,
-  // creates a new encrypted store with the given password,
-  // creates a new HD wallet from the given seed with 1 account.
+  /**
+   * CreateNewVaultAndRestore
+   *
+   * @returns {Promise<Object>} A Promise that resolves to the state.
```

Comment:
Nit: elsewhere the `@returns` declarations were moved to the bottom of the doc comment I noticed